ddtrace/internal/telemetry/writer.py

```diff
@@ -188,10 +187,8 @@ def add_integration(self, integration_name, auto_enabled):
         :param str integration_name: name of patched module
         :param bool auto_enabled: True if module is enabled in _monkey.PATCH_MODULES
         """
-        with self._lock:
-            if self._enabled is not None and not self._enabled:
-                return
-
+        if self._enabled is None or self._enabled:
```

Comment:
Out of curiosity, can you explain why removing the if self._enabled is None resulted in application issues (failed gunicorn tests)? Thanks!

Reply:
In this commit I refactored the `if self._enabled` check to make it more readable (ex: convert `if not enabled: return` to `if enabled: do_work .....`): [183dc254](https://github.com/DataDog/dd-trace-py/pull/5038/commits/69f574c50f77d457b555fe0ce8ce50ea183dc254). In this refactor I forgot that self._enabled boolean is an optional boolean and could be None. This bug prevented integrations from being queued on application start up which caused some ddtrace run test to fail. The failing gunicorn test asserted that a memory allocation error was raised.